### ISSUE TITLE
Missing RightMenu module when try to get app up

### ISSUE THREAD

**Issue by RodrigoCMoraes** [wdsrocha/dojo#23]
First, I needed to install `next`, `react` and `react-dom` to execute `next start` (command executed by scripts at `yarn dev`). I don't know why, but I needed to install globally in order for it works. The command I used:
```shell
npm install -g next react react-dom
```

After installed I got the error below about missing `RightMenu`:

```shell
~/code/dojo/packages/app main*                                                                                                                                                                                                            7s 03:34:42
❯ next build                              
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
warn  - Minimum recommended TypeScript version is v4.3.2, older versions can potentially be incompatible with Next.js. Detected: 3.9.7
Failed to compile.

./components/mainLayout/header/Header.tsx:6:27
Type error: Cannot find module './RightMenu' or its corresponding type declarations.

  4 | import Link from "next/link";
  5 | import { LeftMenu } from "./LeftMenu";
> 6 | import { RightMenu } from "./RightMenu";
    |                           ^
  7 | 
  8 | export const Header = () => {
  9 |   const [isDrawerVisible, setIsDrawerVisible] = useState<boolean>(false);
info  - Checking validity of types .% 
````

**Comment by wdsrocha** [comment]
>  I don't know why, but I needed to install globally in order for it works

tl;dr: run `yarn install` in the `package/app` directory instead.

My bad, I should have added the same step necessary on the `server` package - you need to run `yarn install` (or just `yarn`) to install the dependencies in the `app` package. The `package.json` file (https://github.com/wdsrocha/dojo/blob/f7333c495791267310e09ae562c947d61a79f0af/packages/app/package.json) contains all the required dependencies, so even though `npm install -g next react react-dom` works by installing those libraries globally in your computer, they may be using the incorrect version and also misses the other dependencies (including _dev dependencies_ like the linter, formatter, typescript typings, etc)

More about the `package.json` file: https://nodejs.org/en/knowledge/getting-started/npm/what-is-the-file-package-json/

It's also worthy to mention that `yarn` and `npm` serve the same purpose: manage JavaScript dependencies, but it's not recommended to use both for the same project. One of the reasons for that is because `package.json` _only contains the range of allowed semVer versions_, so for example, here is the version for the `next` package:

https://github.com/wdsrocha/dojo/blob/f7333c495791267310e09ae562c947d61a79f0af/packages/app/package.json#L19

which means any minor and patch version is allowed, but the major shouldn't be different to `10` ([see more about semantic versioning in `package.json`](https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept)).

Although this looks fine, the NextJS team could release a minor or patch version with an unintended breaking change that could break DOJO, so that's why yarn keeps a `yarn.lock` file, which keeps the _exact versions_ that was installed when the developer added new dependencies. [See the official docs for a more in-depth explanation](https://classic.yarnpkg.com/en/docs/yarn-lock/).

`npm` uses `package-lock.json` instead, so if you run `npm install`, it will ignore `yarn.lock` and will create a new `package-lock.json` with possibly different versions. [There are ways to keep them in sync](https://stackoverflow.com/a/44904494/7651928), but for now, I believe making this DOJO strictly dependant on `yarn` is also fine (it's wildly popular in the JS ecosystem, so it shouldn't be a problem for most JS developers).

One last reason is that `yarn workspaces` are used for managing the mono-repository. [More about this](https://classic.yarnpkg.com/en/docs/workspaces/).

**Comment by wdsrocha** [comment]
> After installed I got the error below about missing RightMenu

I hate this. I feel like I've solved this about three times now. In short, the problem is that `macOS` is case insensitive and git is case sensitive, which creates some weird bugs like that one. I've tried following [some tips on SO](https://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git) but it seems that I missed something. I'll use the ultimate git tactic 👀 - delete DOJO from my computer and re-clone it, so I'll be able to see exactly which files git stored.

Thanks for bringing that point up!
